crates/bevy_render/src/lib.rs

```diff
@@ -216,10 +216,54 @@ pub enum RenderSystems {
     PostCleanup,
 }
 
+/// The schedule that contains the app logic that is evaluated each tick
+///
+/// This is highly inspired by [`bevy_app::Main`]
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash, Default)]
+pub struct MainRender;
+impl MainRender {
+    pub fn run(world: &mut World, mut run_at_least_once: Local<bool>) {
+        if !*run_at_least_once {
+            world.resource_scope(|world, order: Mut<MainRenderScheduleOrder>| {
+                for &label in &order.startup_labels {
+                    let _ = world.try_run_schedule(label);
+                }
+            });
+            *run_at_least_once = true;
+        }
+
+        world.resource_scope(|world, order: Mut<MainRenderScheduleOrder>| {
+            for &label in &order.labels {
+                let _ = world.try_run_schedule(label);
+            }
+        });
+    }
+}
+
+/// Defines the schedules to be run for the [`MainRender`] schedule, including
+/// their order.
+#[derive(Resource, Debug)]
+struct MainRenderScheduleOrder {
+    labels: Vec<InternedScheduleLabel>,
+    startup_labels: Vec<InternedScheduleLabel>,
+}
+impl Default for MainRenderScheduleOrder {
+    fn default() -> Self {
+        Self {
+            labels: vec![Render.intern()],
+            startup_labels: vec![RenderStartup.intern()],
```

Comment:
is there a point to these being Vec if there's only ever one label? is this mutable from anywhere? will there be more?

Reply:
MainScheduleOrder exists because Bevy itself wants multiple schedules for its main loop, and users have wanted to extend the schedule order.

Rendering is quite opinionated and users generally don't need to extend it nearly as much (they most likely just want to match the patterns we already have in Bevy). We've already "gotten away" with using a single schedule for everything. I'd rather keep that going and just avoid this resoruce entirely. Let's just hard-code the MainRender::run to call `RenderStartup` and `Render`.